### PR TITLE
feat(ci): run cargo audit in ci

### LINKED ISSUE #431: Update Rocket to 0.4
Ref: https://rocket.rs/v0.4/news/2018-12-08-version-0.4/

### LINKED ISSUE #921: fxa-email-service - run cargo-audit in CI
This project does not appear to be using cargo-audit (https://github.com/RustSec/cargo-audit) - that's strongly recommended in order to be warned about dependencies with security vulnerabilities.

### PR DESCRIPTION
**Pull request by philbooth**
Fixes #921 and [bug 1546710](https://bugzilla.mozilla.org/show_bug.cgi?id=1546710). Fixes #431.

Enables `cargo audit` in CI and wades through the surprisingly hairy process of updating the pinned Rust version and all the deps. Lesson learned, I won't leave it so long next time.

Tested end-to-end, with a real email:

Screenshot of an FxA transactional email sent by the updated email service

@mozilla/fxa-devs r?

### CODE REVIEW

File: packages/fxa-email-service/Cargo.toml [hunk 1]
```diff
@@ -18,27 +18,28 @@ debug = 1
 
 [dependencies]
 base64 = ">=0.6.0"
+bytes = ">=0.4.12"
 chrono = { version = ">=0.4.2", features = [ "serde" ] }
 config = ">=0.9.1"
-emailmessage = ">=0.2.0"
+emailmessage = { git = "https://github.com/mozilla/emailmessage-rs.git", branch = "pb/unicode-local-address-part" }
```
Comment (philbooth) [inline packages/fxa-email-service/Cargo.toml]: Ref: https://github.com/mozilla/fxa/issues/1625

File: packages/fxa-email-service/src/api/send/mod.rs [hunk 2]
```diff
@@ -42,29 +44,44 @@ struct Email {
...
-            .map_failure(|(_status, error)| {
-                (
-                    Status::BadRequest,
-                    AppErrorKind::InvalidPayload(error.to_string()).into(),
-                )
-            })
-            .map(|json| json.into_inner())
+    fn from_data(request: &Request, data: Data) -> Outcome<Self, Self::Error> {
+        let mut email_address = String::new();
+        if let Err(error) = data.open().take(32768).read_to_string(&mut email_address) {
```
Comment (philbooth) [inline packages/fxa-email-service/src/api/send/mod.rs]: Ref: #1628

**Comment by vladikoff** [comment]
awesome, thanks for updating this! r+ per green ci

**Comment by philbooth** [comment]
This seems to have broken the content server tests, which I wasn't expecting:

```
× firefox on linux 4.15.0-1035-aws - email_service - email_service works (45.01s)
    TimeoutError: Timeout reached on firefox on linux 4.15.0-1035-aws - email_service - email_service works#
      at Timeout._onTimeout  <src/lib/Test.ts:221:24>
      at ontimeout  <timers.js:436:11>
      at tryOnTimeout  <timers.js:300:5>
      at listOnTimeout  <timers.js:263:5>
      at Timer.processTimers  <timers.js:223:10>
```

**Comment by vladikoff** [comment]
> This seems to have broken the content server tests, which I wasn't expecting:

Are we missing some headers?

**Comment by philbooth** [comment]
> Are we missing some headers?

Yeah, `X-Email-Service` it seems like. Digging in to it now...

File: packages/fxa-content-server/tests/functional/email_service.js [hunk 3]
```diff
@@ -24,6 +24,7 @@ registerSuite('email_service', {
         .then(fillOutSignUp(email, PASSWORD))
         .then(getEmailHeaders(user, 0))
         .then(headers => {
+          console.log('PHIL!', headers['x-email-service'], headers);
```
Comment (bbangert) [inline packages/fxa-content-server/tests/functional/email_service.js]: 😆

Comment (philbooth) [reply packages/fxa-content-server/tests/functional/email_service.js]: Yeah, debugging it in CI, I should close the PR while I do that sorry!

**Comment by philbooth** [comment]
Temporarily closing while I fix the breakage...

**Comment by philbooth** [comment]
All good now.

**Pull request merged**